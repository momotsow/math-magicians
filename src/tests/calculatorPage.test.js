import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Calculate from '../components/calculator';

test('renders Calculate component', () => {
  const { asFragment } = render(<Calculate />);
  expect(asFragment()).toMatchSnapshot();
});

test('User can click on the buttons', () => {
  const handleClick = jest.fn();
  render(<button type="button" onClick={handleClick} className="input-button">AC</button>);
  const button = screen.getByText('AC');
  userEvent.click(button);
  expect(handleClick).toHaveBeenCalled();
});
