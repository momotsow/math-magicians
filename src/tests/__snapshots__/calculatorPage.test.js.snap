// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculate component 1`] = `
<DocumentFragment>
  <div
    class="homeContainer"
  >
    <div
      class="calculatorContainer"
    >
      <div
        class="desc"
      >
        <h3>
          Let us do some math!
        </h3>
      </div>
      <div
        class="container"
      >
        <div
          class="screen"
        >
          0
        </div>
        <div
          class="buttons"
        >
          <button
            type="button"
          >
            AC
          </button>
          <button
            type="button"
          >
            +/-
          </button>
          <button
            type="button"
          >
            %
          </button>
          <button
            class="orange"
            type="button"
          >
            รท
          </button>
          <button
            type="button"
          >
            7
          </button>
          <button
            type="button"
          >
            8
          </button>
          <button
            type="button"
          >
            9
          </button>
          <button
            class="orange"
            type="button"
          >
            x
          </button>
          <button
            type="button"
          >
            4
          </button>
          <button
            type="button"
          >
            5
          </button>
          <button
            type="button"
          >
            6
          </button>
          <button
            class="orange"
            type="button"
          >
            -
          </button>
          <button
            type="button"
          >
            1
          </button>
          <button
            type="button"
          >
            2
          </button>
          <button
            type="button"
          >
            3
          </button>
          <button
            class="orange"
            type="button"
          >
            +
          </button>
          <button
            class="double"
            type="button"
          >
            0
          </button>
          <button
            type="button"
          >
            .
          </button>
          <button
            class="orange"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <p>
        Developed by
        <a
          href="https://github.com/momotsow/math-magicians"
        >
           KP Mkansi
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
